src/Lean/Elab/App.lean

```diff
@@ -785,9 +785,15 @@ private partial def elabAppFn (f : Syntax) (lvals : List LVal) (namedArgs : Arra
       f.getArgs.foldlM (fun acc f => elabAppFn f lvals namedArgs args expectedType? explicit ellipsis true acc) acc
   else
     let elabFieldName (e field : Syntax) := do
-      let newLVals := field.getId.eraseMacroScopes.components.map fun n =>
-        -- We use `none` here since `field` can't be part of a composite name
-        LVal.fieldName field (toString n) none e
+      let newLVals := 
+        -- We use `none` in `suffix?` since `field` can't be part of a composite name
+        if field.getId.hasMacroScopes then
+          -- We assume that names with macro scopes have no meaningful syntax associated
```

Comment:
I'm not 100% sure this is a correct assumption, is this fine?

Reply:
The problem is that `identComponents` operates on the `rawStr : Substring` field of an `ident`, while in `ident`s with macro scopes, `rawStr` may not match `val : Name`. For example, we have
```lean
deriving instance Repr for Syntax
elab "testme" : command => do
  let stx ← `(foo.test)
  trace[Elab.command] "{repr stx}"

/- [Elab.command] Lean.Syntax.ident (Lean.SourceInfo.synthetic 1064 1084) "foo.test".toSubstring `foo.test._@._hyg.996 [] -/
```
However in the current code we `eraseMacroScopes` anyway. So if it's an invariant that `rawStr.toString == repr val.eraseMacroScopes` then perhaps we can just not use `eraseMacroScopes`. In fact, I can build stage 2 assuming this.

Reply:
Oh well, it builds but the assumption is false. We also have `` Syntax.ident SourceInfo.none "reprSyntax._@._hyg.5".toSubstring `reprSyntax._@._hyg.5 ``. Perhaps the `rawStr` should really be part of `SourceInfo.original` and `val : Name` the sole source of truth.

Reply:
Okay, now I'm assuming the `eraseMacroScopes` representation if original info is present, otherwise using just the `Name`.

Reply:
> Perhaps the `rawStr` should really only be part of `SourceInfo.original` and `val : Name` the sole source of truth in `ident`s.

This is an interesting idea. So it would simply be something like
```
  | original (leading token trailing : Substring)
```
We *might* even be able to replace `Syntax.atom`'s `val : String` with it since at least quotations [completely ignore it](https://github.com/leanprover/lean4/blob/07a4ff2d0fb3704888e2e3857ca563732395b4cc/src/Lean/Elab/Quotation.lean#L268-L269), though this might not be true for all parts of the elaborator (yet). /cc @leodemoura